### DISCUSSION TITLE
build_runner not running codegen in example project

### DISCUSSION THREAD

**Discussion by dragonnn** [fzyzcjy/flutter_rust_bridge#1322] [Q&A] [answered]
Hi! I am testing flutter_rust_bridge and when I add a simple function to api.rs run `flutter pub run build_runner build` bridge_generated.rs is not getting updated and the function can not be run from dart code.
I read the sample book and can not find what I am doing wrong, I am missing some step here?
Also I couldn't find the justfile with is maintained in the book.
Thanks!

**Comment by fzyzcjy** [comment]
Hi, could you please provide a minimal reproducible sample (probably a simple github repo)? And this looks like a bug, so feel free to create an issue!

**Comment by dragonnn** [comment]
Hi, I can make a basic example but I just yesterday cloned the repo and test it with frb_example/with_flutter, add to api.rs this:
```rs
pub fn init() {}
```
run `flutter pub run build_runner build` and nothing have bean picked up. The example itself did run fine before I tried adding my init function and I do have flutter_rust_bridge_codegen installed. I also checked all the steps in https://cjycode.com/flutter_rust_bridge/template/generate.html and nothing did help. Any idea? I will move to a issue if that is one

**Reply by fzyzcjy** [reply]
> run flutter pub run build_runner build and nothing have bean picked up.

this only builds the dart side. you need to execute flutter_rust_bridge codegen

**Comment by dragonnn** [comment]
> 



> > run flutter pub run build_runner build and nothing have bean picked up.
> 
> this only builds the dart side. you need to execute flutter_rust_bridge codegen

Can you show me exactly how? I didn't find any point about how to do it in the guide. I know you can input rust-input and dart-output to flutter_rust_bridge_codegen but I would expect a developer for a project to have to input them manually in each project. Is they any reason why the example project doesn't use build.rs for that witch sounds like the right place to put it?

**Reply by fzyzcjy** [reply, marked as answer]
Oh yes, the doc is a bit stale given that so many new features and new tutorials are added since it was created... Feel free to PR to fix it!

If you like build.rs, https://cjycode.com/flutter_rust_bridge/feature/build_rs.html and https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/frb_example/pure_dart_multi/rust/build.rs (yes I know it is not with_flutter example...).

However, I am not sure whether build.rs is the best way for a Flutter app. You know, the codegen (invoked by build.rs or manually) generates Dart code, and if you run Flutter build which executes build.rs which generates Dart code, you will not get the new Dart code before Flutter build happens, so it may use the old Dart code and be unhappy.

**Reply by dragonnn** [reply]
Thanks! That answers my question, yeah I get the point why build.rs isn't the default, make sense. At least the build.rs example will allow me to make an script that I can run from it to generated. I think it would be helpful if the frb_example project do have an sh file to generate the codegen or a justfile. The guide does say about using just for it https://cjycode.com/flutter_rust_bridge/template/tour_justfile.html but they is no full example of it or a sample in the repo.
For some one first time trying out flutter_rust_bridge it might be hard to wrap they head around what should be put into flutter_rust_bridge_codegen, at least that was for me the case.
Anyway, thanks again! I can now move forward.

**Reply by fzyzcjy** [reply]
You are welcome! I guess we can put something in the `/justfile`. Indeed it should already have something to generate code, since CI is using it.
